### ISSUE TITLE
Bedtools 2.27.1 dependency error

### ISSUE THREAD

**Issue by lzamparo** [bioconda/bioconda-recipes#7202]
Bedtools 2.27.1 from bioconda cannot be installed into a fresh install of Anaconda 5.1, and seems to be due to a malformed dependency string.  Here's me trying to install 2.27.1:

```
lski1877:SeqDemote zamparol$ conda install bedtools=2.27.1
Solving environment: failed

UnsatisfiableError: The following specifications were found to be in conflict:
  - bedtools=2.27.1
  - qtconsole -> pyqt -> dbus[version='>=1.10.22,<2.0a0'] -> glib[version='>=2.53.6,<2.54.0a0'] -> zlib[version='>=1.2.11,<1.3.0a0']
```
But 2.26 can install just fine:
```
lski1877:SeqDemote zamparol$ conda install bedtools=2.26
Solving environment: done

## Package Plan ##

  environment location: /Users/zamparol/anaconda36

  added / updated specs:
    - bedtools=2.26


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    bedtools-2.26.0            |                0         645 KB  bioconda

The following NEW packages will be INSTALLED:

    bedtools: 2.26.0-0 bioconda
```

Now look at the dependencies of 2.27.1

```
lski1877:SeqDemote zamparol$ conda info bedtools=2.27.1

bedtools 2.27.1 0
-----------------
[snip]
dependencies:
    zlib 1.2.8*
```
 versus 2.26:

```
lski1877:SeqDemote zamparol$ conda info bedtools=2.26

bedtools 2.26.0 0
-----------------
[snip]
dependencies:
    zlib
```

Is it possible to amend the formatting of the dependency string so that I can install this version of bedtools (which I need, has the critical groupby bugfix)?  

Thanks,

**Comment by cbrueffer** [comment]
The problem seems to be rather that the ```qtconsole``` dependency chain needs a ```zlib``` newer than what bioconda currently uses: ```zlib[version='>=1.2.11,<1.3.0a0']```.

**Comment by cbrueffer** [comment]
To expand on this, newer bioconda packages are all built against the same zlib version (1.2.8), including ```bedtools 2.27.1```. The ```bedtools 2.26.0``` package was built before that zlib version was fixed, so it doesn't have a strict dependency on zlib 1.2.8. When installing bedtools now, conda wants to downgrade your zlib which is incompatible with what the glib package requires.

Not sure what the best way to fix this is; I suppose you could try installing an version of glib.

**Comment by mbargull** [comment]
@lzamparo, `conda` should figure out that it needs to adjust/downgrade other packages like `glib`/`zlib` when you run `conda install bedtools=2.27.1`. I cannot see why it does not in your case.
Can you give more information on your setup? E.g., outputs from
* `conda info`
* `conda config --show-sources`
* `conda config --show`
* `conda list`

**Comment by lzamparo** [comment]
@cbrueffer thanks for finding the cause.  The reward from installing a more fundamental package (glib) to install bedtools seems disproportionate to the work involved; is it not possible to change the configuration of bedtools to depend on zlib >= 1.2.8 instead?  Or do you need the stability of building against only 1.2.8?

**Comment by lzamparo** [comment]
@mbargull, I think @cbrueffer found the reason, it's that a downgrade of `zlib` to 1.2.8 isn't possible since it conflicts with the `qtconsole` dependency chain requirement on `zlib >= 1.2.11`

**Comment by mbargull** [comment]
That `zlib` dependency is the reason of the conflict, yes. But `conda` should be able to resolve that conflict by downgrading the other packages. Apparently there is a bug in `conda 4.4.x` regarding update specs or such in the `base`/`root` environment.
I'm able to reproduce your issue with `conda 4.4.7` (but not with `conda 4.3.31`) and having `anaconda` installed in `base`. If you install `anaconda` into a separate environment (which I'd recommend anyways), it works just fine:
```bash
conda create -n anaconda anaconda
conda install -n anaconda bedtools=2.27.1
```
I will see if I can create a MWE (which doesn't include all of `anaconda`) and open an acompaning issue for `conda`.

**Comment by lzamparo** [comment]
Ah, understood.  Thanks for following up on this.

**Comment by mbargull** [comment]
This is also fixed upstream with `conda 4.4.11` via https://github.com/conda/conda/pull/6921.
